src/components/panel/panel.spec.js

```diff
@@ -96,6 +68,139 @@ describe('$mdPanel', function() {
     expect(PANEL_EL).not.toExist();
   });
 
+  it('should hide and show a panel in the DOM', function() {
+    var template = '<div>Hello World!</div>';
+    var config = { template: template };
+
+    openPanel(config);
+
+    expect(PANEL_EL).toExist();
+    expect(PANEL_WRAPPER_CLASS).not.toHaveClass(HIDDEN_CLASS);
+
+    hidePanel();
+
+    expect(PANEL_EL).toExist();
+    expect(PANEL_WRAPPER_CLASS).toHaveClass(HIDDEN_CLASS);
+
+    showPanel();
+
+    expect(PANEL_EL).toExist();
+    expect(PANEL_WRAPPER_CLASS).not.toHaveClass(HIDDEN_CLASS);
+  });
+
+  describe('promises logic:', function() {
+    var config;
+
+    beforeEach(function() {
+      config = {};
+
+      panelRef = $mdPanel.create(config);
+      expect(panelRef.isAttached).toEqual(false);
+    });
+
+    it('should resolve when opening/closing', function () {
+      var openResolved = false;
+      var closeResolved = false;
+
+      expect(panelRef.id).toBeDefined();
+      expect(panelRef.open).toBeOfType('function');
+      expect(panelRef.close).toBeOfType('function');
```

Comment:
This message was created automatically by mail delivery software.

A message that you sent could not be delivered to one or more of its
recipients. This is a temporary error. The following address(es) deferred:

  dupuyyann@gmail.com
    Domain zyapps.com has exceeded the max emails per hour (417/400 (104%)) allowed.  Message will be reattempted later

------- This is a copy of the message, including all the headers. ------
Received: from github-smtp2-ext3.iad.github.net ([192.30.252.194]:53732 helo=github-smtp2b-ext-cp1-prd.iad.github.net)
	by serveur.ex2-p10.com with esmtps (TLSv1.2:ECDHE-RSA-AES256-GCM-SHA384:256)
	(Exim 4.86_1)
	(envelope-from <noreply@github.com>)
	id 1afmMd-000ajn-V1
	for yann.dupuy@zyapps.com; Tue, 15 Mar 2016 11:38:22 +0100
Date: Mon, 14 Mar 2016 13:30:39 -0700
DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed; d=github.com;
	s=pf2014; t=1457987439;
	bh=utXgOOJgFx8VFH9hX/UXFueY/iKH8nbYhrftdy1YiXg=;
	h=From:Reply-To:To:In-Reply-To:References:Subject:List-ID:
	 List-Archive:List-Post:List-Unsubscribe:From;
	b=nQzTYZRk1OWCbcSNN+zPF6kz5owMx2LFcM8ptZsMn35MRctpOPPPhQjx/Bd/DIozK
	 eO9qdzKZNkBgq2wHK/n+t3/SdgaQFj7RUsPV5X86OZa+tPRt75prkMv5AulOB5WxzO
	 f2XOXlP/RHfRH5oC1AK+BhO7p9KHGCR8RAleMRYA=
From: Karen Parker <notifications@github.com>
Reply-To: angular/material <reply@reply.github.com>
To: angular/material <material@noreply.github.com>
Message-ID: <angular/material/pull/7461/r56070238@github.com>
In-Reply-To: <angular/material/pull/7461@github.com>
References: <angular/material/pull/7461@github.com>
Subject: Re: [material] update(panel): Add show and hide to mdPanelRef.
 Cleaned up open and close. (#7461)
Mime-Version: 1.0
Content-Type: multipart/alternative;
 boundary="--==_mimepart_56e71f6f724ae_4e263fea655fd2c02001ea";
 charset=UTF-8
Content-Transfer-Encoding: 7bit
Precedence: list
X-GitHub-Sender: KarenParker
X-GitHub-Recipient: yanndupuy
List-ID: angular/material <material.angular.github.com>
List-Archive: https://github.com/angular/material
List-Post: <mailto:reply@reply.github.com>
List-Unsubscribe: <mailto:unsub+000df0dc886eda5bfd14c44335796adf84d8c7c96e3e5ea092cf0000000112fee16f92a170ce0357905e@reply.github.com>,
 <https://github.com/notifications/unsubscribe/AA3w3E9lyyD6f-eLv5AJxx1qXx92AUqJks5ptcVvgaJpZM4Hr7Ku>
X-Auto-Response-Suppress: All
X-GitHub-Recipient-Address: yann.dupuy@zyapps.com
X-Spam-Status: No, score=-3.2
X-Spam-Score: -31
X-Spam-Bar: ---
X-Ham-Report: Spam detection software, running on the system "serveur.ex2-p10.com",
 has NOT identified this incoming email as spam.  The original
 message has been attached to this so you can view it or label
 similar future email.  If you have any questions, see
 root\@localhost for details.
 
 Content preview:  > + var config; > + > + beforeEach(function() { > + config
    = {}; > + > + panelRef = $mdPanel.create(config); > + expect(panelRef.isAttached).toEqual(false);
    > + }); > + > + it('should resolve when opening/closing', function () { >
    + var openResolved = false; > + var closeResolved = false; > + > + expect(panelRef.id).toBeDefined();
    > + expect(panelRef.open).toBeOfType('function'); > + expect(panelRef.close).toBeOfType('function');
    [...] 
 
 Content analysis details:   (-3.2 points, 5.0 required)
 
  pts rule name              description
 ---- ---------------------- --------------------------------------------------
 -5.0 RCVD_IN_DNSWL_HI       RBL: Sender listed at http://www.dnswl.org/, high
                             trust
                             [192.30.252.194 listed in list.dnswl.org]
 -0.0 RCVD_IN_MSPIKE_H2      RBL: Average reputation (+2)
                             [192.30.252.194 listed in wl.mailspike.net]
  0.8 DATE_IN_PAST_12_24     Date: is 12 to 24 hours before Received: date
 -0.0 SPF_PASS               SPF: sender matches SPF record
  0.0 HTML_MESSAGE           BODY: HTML included in message
  1.0 HTML_IMAGE_ONLY_16     BODY: HTML: images with 1200-1600 bytes of words
 -0.1 DKIM_VALID_AU          Message has a valid DKIM or DK signature from author's
                             domain
  0.1 DKIM_SIGNED            Message has a DKIM or DK signature, not necessarily valid
 -0.1 DKIM_VALID             Message has at least one valid DKIM or DK signature
X-Spam-Flag: NO


----==_mimepart_56e71f6f724ae_4e263fea655fd2c02001ea
Content-Type: text/plain;
 charset=UTF-8
Content-Transfer-Encoding: 7bit

> +    var config;
> +
> +    beforeEach(function() {
> +      config = {};
> +
> +      panelRef = $mdPanel.create(config);
> +      expect(panelRef.isAttached).toEqual(false);
> +    });
> +
> +    it('should resolve when opening/closing', function () {
> +      var openResolved = false;
> +      var closeResolved = false;
> +
> +      expect(panelRef.id).toBeDefined();
> +      expect(panelRef.open).toBeOfType('function');
> +      expect(panelRef.close).toBeOfType('function');

This code was here before, I just moved it into this describe.

---
Reply to this email directly or view it on GitHub:
https://github.com/angular/material/pull/7461/files#r56070238
----==_mimepart_56e71f6f724ae_4e263fea655fd2c02001ea
Content-Type: text/html;
 charset=UTF-8
Content-Transfer-Encoding: 7bit

<p>In <a href="https://github.com/angular/material/pull/7461#discussion_r56070238">src/components/panel/panel.spec.js</a>:</p>
<pre style='color:#555'>&gt; +    var config;
&gt; +
&gt; +    beforeEach(function() {
&gt; +      config = {};
&gt; +
&gt; +      panelRef = $mdPanel.create(config);
&gt; +      expect(panelRef.isAttached).toEqual(false);
&gt; +    });
&gt; +
&gt; +    it(&#39;should resolve when opening/closing&#39;, function () {
&gt; +      var openResolved = false;
&gt; +      var closeResolved = false;
&gt; +
&gt; +      expect(panelRef.id).toBeDefined();
&gt; +      expect(panelRef.open).toBeOfType(&#39;function&#39;);
&gt; +      expect(panelRef.close).toBeOfType(&#39;function&#39;);
</pre>
<p>This code was here before, I just moved it into this describe.</p>

<p style="font-size:small;-webkit-text-size-adjust:none;color:#666;">&mdash;<br>Reply to this email directly or <a href="https://github.com/angular/material/pull/7461/files#r56070238">view it on GitHub</a>.<img alt="" height="1" src="https://github.com/notifications/beacon/AA3w3KegQ2aqe1UykS0b71-o1xQPqp3yks5ptcVvgaJpZM4Hr7Ku.gif" width="1" /></p>
<div itemscope itemtype="http://schema.org/EmailMessage">
<div itemprop="action" itemscope itemtype="http://schema.org/ViewAction">
  <link itemprop="url" href="https://github.com/angular/material/pull/7461/files#r56070238"></link>
  <meta itemprop="name" content="View Pull Request"></meta>
</div>
<meta itemprop="description" content="View this Pull Request on GitHub"></meta>
</div>

----==_mimepart_56e71f6f724ae_4e263fea655fd2c02001ea--